### PR TITLE
Check for liveness of process on nix-repl completions

### PR DESCRIPTION
**Pull request by rossabaker**
Suggested in review of #107.

### CODE REVIEW

File: nix-repl.el [hunk 1]
```diff
@@ -149,6 +149,7 @@ guarantees they will be grabbed in a single call."
 See `completion-at-point-functions'."
   (save-excursion
     (let* ((proc (get-buffer-process (current-buffer)))
+           (process-live-p proc)
```
Comment (purcell) [inline nix-repl.el:152]: I don't think this does what you think... :D

File: nix-repl.el [hunk 2]
```diff
@@ -149,6 +149,7 @@ guarantees they will be grabbed in a single call."
...
            (prefix (and (derived-mode-p 'nix-repl-mode)
                         proc
```
Comment (purcell) [inline nix-repl.el:154]:
```suggestion
                        (process-live-p proc)
```

**Review by purcell — CHANGES_REQUESTED** [review]

File: nix-repl.el (hunk 1 above)
Comment (rossabaker) [reply nix-repl.el:152]: Egads, that was a silly mistake. Thank you.

**Review by rossabaker — COMMENTED** [review]
@purcell is of course correct. But after consulting the docs, I think we might just be able to close this:

```
Signature
(get-buffer-process BUFFER)

Documentation
Return the (or a) live process associated with BUFFER.
```

Unless `process-live-p` is more stringent than the liveness check in `get-process-buffer`, all we're guarding against is an unlikely crash while we're checking the mode, no?

**Comment by purcell** [comment]
> Unless `process-live-p` is more stringent than the liveness check in `get-process-buffer`, all we're guarding against is an unlikely crash while we're checking the mode, no?

Yeah, guess so.

**Pull request closed without merging**